src/Compilers/Core/Portable/Collections/TopologicalSort.cs

```diff
@@ -62,14 +63,11 @@ public static ImmutableArray<TNode> IterativeSort<TNode>(IEnumerable<TNode> node
             }
 
             // At this point all the nodes should have been output, otherwise there was a cycle
-            if (predecessorCounts.Count != resultBuilder.Count)
-            {
-                throw new ArgumentException("Cycle in the input graph");
-            }
-
+            bool hadCycle = predecessorCounts.Count != resultBuilder.Count;
+            result = hadCycle ? ImmutableArray<TNode>.Empty : resultBuilder.ToImmutable();
             predecessorCounts.Free();
             ready.Free();
```

Comment:
Looks like there is a missing `resultBuilder.Free();` here.

Reply:
It isn't pooled.

---
In reply to: [455509300](https://github.com/dotnet/roslyn/pull/45983#discussion_r455509300) [](ancestors = 455509300)